app/src/main/java/com/lms/mpasho_lms_news/Adapter.java

```diff
@@ -0,0 +1,130 @@
+package com.lms.mpasho_lms_news;
+
+import android.content.Context;
+import android.content.DialogInterface;
+import android.graphics.drawable.Drawable;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ImageView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.load.DataSource;
+import com.bumptech.glide.load.engine.DiskCacheStrategy;
+import com.bumptech.glide.load.engine.GlideException;
+import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;
+import com.bumptech.glide.request.RequestListener;
+import com.bumptech.glide.request.RequestOptions;
+import com.bumptech.glide.request.target.Target;
+import com.lms.mpasho_lms_news.models.Article;
+
+import java.util.List;
+
+
+public class Adapter extends RecyclerView.Adapter<Adapter.MyViewHolder> {
+
+    private List<Article> articles;
+    private Context context;
+    private OnItemClickListener onItemClickListener;
+
+
+    public Adapter(List<Article> articles, Context context) {
+        this.articles = articles;
+        this.context = context;
+    }
+
+    @NonNull
+    @Override
+    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
+        View view = LayoutInflater.from(context).inflate(R.layout.item, viewGroup, false);
+        return new MyViewHolder(view, onItemClickListener);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull MyViewHolder viewHolder, int position) {
+        final MyViewHolder holder = viewHolder;
+        Article model = articles.get(position);
+
+        RequestOptions requestOptions = new RequestOptions();
+        requestOptions.placeholder(Utils.getRandomDrawbleColor());
+        requestOptions.error(Utils.getRandomDrawbleColor());
+        requestOptions.diskCacheStrategy(DiskCacheStrategy.ALL);
+        requestOptions.centerCrop();
+
+        Glide.with(context)
+                .load(model.getUrlToImage())
+                .apply(requestOptions)
+                .listener(new RequestListener<Drawable>() {
+                    @Override
+                    public boolean onLoadFailed(@Nullable GlideException e, Object model, Target<Drawable> target, boolean isFirstResource) {
+                        holder.progressBar.setVisibility(View.GONE);
+                        return false;
+                    }
+
+                    @Override
+                    public boolean onResourceReady(Drawable resource, Object model, Target<Drawable> target, DataSource dataSource, boolean isFirstResource) {
+                        holder.progressBar.setVisibility(View.GONE);
+                        return false;
```

Comment:
You can add a way to toast to the user, the reason their page failed to load. Right?

Reply:
Not really, this loads the image for the single news item, if the image doesn't load, it shows an empty  view for the iage